grails-app/i18n/messages.properties

```diff
@@ -655,6 +654,13 @@ documents.label=Documents
 documentType.label=Document Type
 documentTypes.label=Document Types
 
+# Date formats
+default.date.format=dd/MMM/yyyy hh:mm:ss a z
+common.date.format=MM/dd/yyyy
+expiry.date.format=MM/yyyy
+fullmonth.date.format=MMMM dd, yyyy
+abbreviatedmonth.date.format=dd/MMM/yyyy
```

Comment:
As I mentioned, I don't like this convention. It's too specific and you've broken the convention immediately upon adding the date formats to es-MX.
```
fullmonth.date.format=dd/MMM/yyyy
```

Is there a better way of doing this that allows us to choose a format by type/style, while also being non-committal on the actual implementation?

It's a built in feature using type and style attributes rather than formatName. 
```
<g:formatDate date="${date}" type="date" style="SHORT"/>
<g:formatDate date="${date}" type="time" style="LONG"/>
<g:formatDate date="${date}" type="datetime" style="MEDIUM"/>
<g:formatDate date="${date}" type="datetime" style="LONG" timeStyle="SHORT"/>
```

So if we're going down this path, then let's use type and style and then you get to go with type/style convention while also being somewhat customizable. It limits the combinations allowed, but gives us a rigidity / convention that I think we're looking for. 

Here's the path forward. We have two options. 

1. we hard-code MMM into all date formats and leave this larger refactoring for later
1. we continue down the refactoring path and we use a combination of the following, but use type and style for most date, time, and datetime formatting
    * use type and style format almost everything
        * type=date,time,datetime
        * style=short,medium,long
    * use format name for things where there's a pretty standard convention based on purpose
        * expiry.date.format (MMM/yyyy)
        * auditing.date.format ![image](https://github.com/openboxes/openboxes/assets/1162935/5e1614b1-2436-4e60-a098-e202a2f78f2d)
        * auditing.datetime.format ![image](https://github.com/openboxes/openboxes/assets/1162935/bb20aa97-82ba-4183-ade0-668f58bf64a3)
    * use neither accept the default for all other dates where the default format is ok